libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/ReplayLocationEngine.kt

```diff
@@ -106,7 +106,7 @@ class ReplayLocationEngine(
         val locationEngineResult = LocationEngineResult.create(location)
         lastLocationEngineResult = locationEngineResult
 
-        registeredCallbacks.forEach { it.onSuccess(locationEngineResult) }
+        registeredCallbacks.toTypedArray().forEach { it.onSuccess(locationEngineResult) }
```

Comment:
```suggestion
        registeredCallbacks.toTypedArray().forEach { it.onSuccess(locationEngineResult) }
```
We generally use `CopyOnWriteArrayList`s to support this use case across the SDK, without having to remember to copy the list on each invocation. Could you refactor the field itself?

Reply:
sure thing. It was a second option